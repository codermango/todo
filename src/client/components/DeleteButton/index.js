import React from 'react';
import PropTypes from 'prop-types';

const DeleteButton = ({ onDelete }) => (
  <button onClick={onDelete} data-test-id="delete-button">X</button>
);

DeleteButton.propTypes = {
  onDelete: PropTypes.func.isRequired
};

export default DeleteButton;
