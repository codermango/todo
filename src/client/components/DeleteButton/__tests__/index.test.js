import React from 'react';
import { shallow } from 'enzyme';
import DeleteButton from '../index';


describe('<DeleteButton />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<DeleteButton onDelete={jest.fn()} />);
    const instance = wrapper.html();
    expect(instance).toMatchSnapshot();
  });

  it('onClick is called when delete button is clicked', () => {
    const wrapper = shallow(<DeleteButton onDelete={jest.fn()} />);
    wrapper.find('button').simulate('click');
    expect(wrapper.props().onClick).toHaveBeenCalledTimes(1);
  });
});
