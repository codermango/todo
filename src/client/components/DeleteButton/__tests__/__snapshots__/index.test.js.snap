// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DeleteButton /> renders correctly 1`] = `"<button data-test-id=\\"delete-button\\">X</button>"`;
